app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -4,91 +4,86 @@
 
 package org.mozilla.fenix.wallpapers
 
-import android.content.Context
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 import mozilla.components.concept.fetch.Client
 import mozilla.components.concept.fetch.Request
 import mozilla.components.concept.fetch.isSuccess
-import mozilla.components.support.base.log.logger.Logger
 import org.mozilla.fenix.BuildConfig
+import org.mozilla.fenix.wallpapers.Wallpaper.Companion.getLocalPath
 import java.io.File
+import java.lang.IllegalStateException
 
 /**
  * Can download wallpapers from a remote host.
  *
- * @param context Required for writing files to local storage.
+ * @param filesDir The top level app-local storage directory.
  * @param client Required for fetching files from network.
+ * @param dispatcher Dispatcher used to execute suspending functions. Default parameter
+ * should be likely be used except for when under test.
  */
 class WallpaperDownloader(
-    private val context: Context,
+    private val filesDir: File,
     private val client: Client,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
 ) {
-    private val logger = Logger("WallpaperDownloader")
     private val remoteHost = BuildConfig.WALLPAPER_URL
 
     /**
-     * Downloads a wallpaper from the network. Will try to fetch 4 versions of each wallpaper:
-     * portrait/light - portrait/dark - landscape/light - landscape/dark. These are expected to be
-     * found at a remote path in the form:
-     * <WALLPAPER_URL>/<resolution>/<orientation>/<app theme>/<wallpaper theme>/<wallpaper name>.png
+     * Downloads a wallpaper from the network. Will try to fetch 2 versions of each wallpaper:
+     * portrait and landscape. These are expected to be found at a remote path in the form:
+     * <WALLPAPER_URL>/<collection name>/<wallpaper name>/<orientation>.png
+     * and will be stored in the local path:
+     * wallpapers/<wallpaper name>/<orientation>.png
      */
-    suspend fun downloadWallpaper(wallpaper: Wallpaper) = withContext(Dispatchers.IO) {
-        if (remoteHost.isNullOrEmpty()) {
-            return@withContext
+    suspend fun downloadWallpaper(wallpaper: Wallpaper) = withContext(dispatcher) {
+        listOf(Wallpaper.ImageType.Portrait, Wallpaper.ImageType.Landscape).map { imageType ->
+            wallpaper.downloadAsset(imageType)
         }
```

Comment:
This would additionally download the thumbnails, which should have already happened. I could update it to be something like:
```
val portraitResult = wallpaper.downloadAsset(Wallpaper.ImageType.Portrait)
val landscapeResult = wallpaper.downloadAsset(Wallpaper.ImageType.Landscape)
return if (portraitResult = Wallpaper.ImageFileState.Downloaded && landscapeResult = Wallpaper.ImageFileState.Downloaded) {
    Wallpaper.ImageFileState.Downloaded
} else {
    Wallpaper.ImageFileState.Error
}
```
This would also stop the call sites from needing to perform the logic to check both results.

Reply:
this looks good to me